src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs

```diff
@@ -125,4 +125,55 @@ sealed class DoesNotReturnIfAttribute : Attribute
         /// <summary>Gets the condition parameter value.</summary>
         public bool ParameterValue { get; }
     }
+
+    /// <summary>Specifies that the method or property will ensure that the listed field and property members have not-null values.</summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, Inherited = false)]
+#if INTERNAL_NULLABLE_ATTRIBUTES
+    internal
+#else
+    [CLSCompliant(false)]
+    public
+#endif
+        sealed class MemberNotNullAttribute : Attribute
+    {
+        /// <summary>Gets field or property member names.</summary>
+        public string[] Members { get; }
```

Comment:
@terrajobst, concerns?

Reply:
@terrajobst Just confirming that it's okay to add the properties to store the arguments to the attribute. This is following what we did for other nullability attributes.

Reply:
Agreed. `NullableAttribute` also [exposes a raw byte array](https://github.com/dotnet/runtime/issues/29039#issuecomment-498481064). /cc @bartonjs